File: finger_exercises/chapter3/pwr_root.py
"""Write a program that asks the user to enter an integer
and prints two integers, root and pwr such that 1< pwr < 6 and root**pwr == integer"""

x = int(input("Enter an integer: "))

ans = None
for i in range(1, x):
    if ans != None:
        break
    for j in range(2,6):
        if i**j == x:
            ans = (i, j)
            break
if ans != None:
    print(f"Root = {ans[0]}, Power = {ans[1]}")
else:
    print("No answer found!")

File: finger_exercises/chapter3/pwr_root.c
#include <stdio.h>
#include <math.h>

int main(void){
    int x;
    int pwr;
    int root;
    
    printf("Enter an integer: ");
    scanf("%d", &x);

    for (root=2; root < x; root++)
    {
        for (pwr = 2; pwr < 6; pwr++)
        {
            if (pow((double)root, (double)pwr) == x)
            {
                printf("Root = %d, Power = %d.\n", root, pwr);
                return 0;
            }
        }
    }
    
    printf("No combination found\n.");
    
    return 0;
}